src/code/z_player_lib.c

```diff
@@ -723,23 +723,34 @@ void Player_UpdateBottleHeld(PlayState* play, Player* this, s32 item, s32 itemAc
     this->itemAction = itemAction;
 }
 
-void func_8008EDF0(Player* this) {
+void Player_ReleaseLockOn(Player* this) {
     this->focusActor = NULL;
     this->stateFlags2 &= ~PLAYER_STATE2_LOCK_ON_WITH_SWITCH;
 }
 
-void func_8008EE08(Player* this) {
+/**
+ * This function aims to clear Z-Target related state when it isn't in use.
+ * First, many conditions related to being in the air are checked. If these all pass
+ * then many Z-Target related state flags can be unset.
+ * If the long chain of conditions failed, then Player is considered to be in the air.
+ * In this case, `PLAYER_STATE1_FREE_FALL` will be set instead of Z-Target state flags being cleared.
+ * Regardless of the free fall state, actor lock-on will unconditionally be able to release.
+ *
+ * This strange relationship between Z-Targeting and free fall state is why it is not possible
+ * to release parallel mode while in the air.
+ */
+void Player_ClearZTargeting(Player* this) {
     if ((this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) ||
         (this->stateFlags1 & (PLAYER_STATE1_21 | PLAYER_STATE1_23 | PLAYER_STATE1_27)) ||
-        (!(this->stateFlags1 & (PLAYER_STATE1_18 | PLAYER_STATE1_19)) &&
+        (!(this->stateFlags1 & (PLAYER_STATE1_18 | PLAYER_STATE1_FREE_FALL)) &&
          ((this->actor.world.pos.y - this->actor.floorHeight) < 100.0f))) {
         this->stateFlags1 &= ~(PLAYER_STATE1_Z_TARGETING | PLAYER_STATE1_FRIENDLY_ACTOR_FOCUS | PLAYER_STATE1_PARALLEL |
-                               PLAYER_STATE1_18 | PLAYER_STATE1_19 | PLAYER_STATE1_LOCK_ON_FORCED_TO_RELEASE);
-    } else if (!(this->stateFlags1 & (PLAYER_STATE1_18 | PLAYER_STATE1_19 | PLAYER_STATE1_21))) {
-        this->stateFlags1 |= PLAYER_STATE1_19;
+                               PLAYER_STATE1_18 | PLAYER_STATE1_FREE_FALL | PLAYER_STATE1_LOCK_ON_FORCED_TO_RELEASE);
+    } else if (!(this->stateFlags1 & (PLAYER_STATE1_18 | PLAYER_STATE1_FREE_FALL | PLAYER_STATE1_21))) {
+        this->stateFlags1 |= PLAYER_STATE1_FREE_FALL;
```

Comment:
I suppose this matches the cam mode name, but maybe something like "untargeted free fall" might be better? From the name I would guess this flag is set whenever you're falling, but IIUC this flag is not set if you're falling but targeted since Player_ClearZTargeting won't be called? The "untargeted" part might be good to let the reader know about, since maybe you can dodge some of the checks for this flag by staring targeted?

Reply:
I feel like we have more to learn about this state/flag. I cant get it to activate outside of intiating free fall from a debug no-clip, and for like one frame when finishing climbing up something.

We can try to investigate it further, but it might be worth keeping it unnamed for the time being

https://github.com/user-attachments/assets/bc935802-d9b3-4cbb-9d88-8099ec56cebe

Reply:
well it seems like "jumping" (including sidehop/backflip) won't set the flag so your first two examples make sense. Maybe some other ways to become airborne are walking off a slope with no jump, getting pushed off, having the floor disappear, or getting knockback

would be fine not naming it in this PR though

Reply:
yeah, think we will just need a different term for this since freefall feels too broad. Ill add a TODO in the function comment I guess to write more details later